src/field_theory/minimal_polynomial.lean

```diff
@@ -65,6 +65,131 @@ ne_zero_of_monic (monic hx)
 
 end ring
 
+section integral_domain
+
+variables [integral_domain α]
+
+section ring
+
+variables [ring β] [algebra α β] [nontrivial β]
+variables {x : β} (hx : is_integral α x)
+
+/--The degree of a minimal polynomial is nonzero.-/
+lemma degree_ne_zero : degree (minimal_polynomial hx) ≠ 0 :=
```

Comment:
In fact, the degree is positive (using `ne_zero` and `zero_le_degree_iff`) and we don't even have to assume `α` is an integral domain. So in the `ring` section we can add:

```lean
/--The degree of a minimal polynomial is positive. -/
lemma degree_pos [nontrivial α] [nontrivial β] : 0 < degree (minimal_polynomial hx) :=
begin
  apply lt_of_le_of_ne,
  { simpa only [zero_le_degree_iff] using ne_zero hx },
  assume deg_eq_zero,
  rw eq_comm at deg_eq_zero,
  have ndeg_eq_zero : nat_degree (minimal_polynomial hx) = 0,
  { simpa using congr_arg nat_degree (eq_C_of_degree_eq_zero deg_eq_zero) },
  have eq_one : minimal_polynomial hx = 1,
  { rw eq_C_of_degree_eq_zero deg_eq_zero, convert C_1,
    simpa only [ndeg_eq_zero.symm] using (monic hx).leading_coeff },
  simpa only [eq_one, alg_hom.map_one, one_ne_zero] using aeval hx
end
```